File: datasets/lua/HumanEval_11_string_xor.lua
-- Input are two strings a and b consisting only of 1s and 0s.
-- Perform binary XOR on these inputs and return result also as a string.
-- >>> string_xor('010', '110')
-- '100'
local function string_xor(a, b)
  local result = ''
  for i = 1, #a do
    local x = a:sub(i, i)
    local y = b:sub(i, i)
    if x == y then
      result = result .. '0'
    else
      result = result .. '1'
    end
  end
  return result
end
-- Unit tests below

lu = require('luaunit')

function test_humaneval()
local candidate = string_xor
    lu.assertEquals(candidate('111000', '101010'), '010010')
    lu.assertEquals(candidate('1', '1'), '0')
    lu.assertEquals(candidate('0101', '0000'), '0101')
end

os.exit(lu.LuaUnit.run())

File: datasets/rb/HumanEval_11_string_xor.rb
# Input are two strings a and b consisting only of 1s and 0s.
# Perform binary XOR on these inputs and return result also as a string.
# >>> string_xor('010', '110')
# '100'
def string_xor(a, b)
  a.to_i(2) ^ b.to_i(2).to_s(2)
end

require 'test/unit'
class TestHumanEval < Test::Unit::TestCase
  def test_string_xor
    candidate = method(:string_xor)
    assert_equal('010010', candidate.call('111000', '101010'))
    assert_equal('0', candidate.call('1', '1'))
    assert_equal('0101', candidate.call('0101', '0000'))
  end
end
